compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUCastTypeToFitMMA.cpp

```diff
@@ -103,14 +133,14 @@ struct LLVMGPUCastTypeToFitMMAPass final
       }
     }
 
-    // Import mma type from dispatch schedule attribute if present.
-    if (scheduleAttr) {
-      func.walk([&](vector::ContractionOp contract) {
-        if (!contract->hasAttr("iree.amdgpu.mma")) {
-          contract->setAttr("iree.amdgpu.mma", scheduleAttr.getIntrinsic());
-        }
-      });
-    }
+    // Set MMA type from config embedded in toLayoutOp of contraction.
+    // If none, use mma type from dispatch schedule attribute if present.
+    func.walk([&](vector::ContractionOp contract) {
+      inferMmaKind(contract);
+      if (!contract->hasAttr("iree.amdgpu.mma") && scheduleAttr) {
+        contract->setAttr("iree.amdgpu.mma", scheduleAttr.getIntrinsic());
+      }
+    });
```

Comment:
We should completely remove the scheduleAttr dependency here. inferring should be enough.

Reply:
Defo can remove, but was thinking about that, are we sure that no one else relies on this feature? we can still remove but probably should notify people.

Reply:
I doubt anyone is using this pass other than VectorDistribute pipeline. We can go ahead and delete it. The next patch will remove MMASchedule attr completly.

Reply:
sure :)

Reply:
done!